### ISSUE TITLE
steam pipe fails to work when using aws sso credentials

### ISSUE THREAD

**Issue by thapabishwa** [turbot/steampipe#242]
My organization recently made the switch from aws iam user to aws sso credentials(compliance reasons). As such, when using sso credentials steampipe fails to work.
```
~ >>> aws sso login --profile myprofile                                                                                                                           
Attempting to automatically open the SSO authorization page in your default browser.
If the browser does not open or you wish to use a different device to authorize this request, open the following URL:

https://device.sso.us-west-2.amazonaws.com/

Then enter the code:

SAMPLE-CODE
Opening in existing browser session.
Successully logged into Start URL: https://sampleurl.awsapps.com/start#/
~ >>> steampipe query                                                                                                                                                       
Welcome to Steampipe v0.2.2
For more information, type .help
> select * from aws_account
Error: pq: rpc error: code = Unknown desc = NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
>
```

**Comment by LalitTurbot** [comment]
Thanks, @thapabishwa for trying out Steampipe

I am trying to replicate the issue on my side and resolve it in the upcoming version of https://github.com/turbot/steampipe-plugin-aws plugin

**Comment by thapabishwa** [comment]
Thanks for letting me know that it is in the roadmap @LalitTurbot. Let me know if I can help you out in anyways.

**Comment by e-gineer** [comment]
Hey @thapabishwa ... Under the hood the aws sso just creates a temporary credential set. These are often also available through the AWS SSO UI. [AWS Docs](https://aws.amazon.com/blogs/security/aws-single-sign-on-now-enables-command-line-interface-access-for-aws-accounts-using-corporate-credentials/).

So until we have direct support in steampipe, you can consider using steampipe's [connection management](https://hub.steampipe.io/plugins/turbot/aws#configuration-arguments) to set those temporary credentials? Hopefully that will unblock you.

**Comment by thapabishwa** [comment]
Thanks for that @e-gineer . It would at-least unblock me for a while.

**Comment by cbruno10** [comment]
Hi @thapabishwa, this should be fixed in the latest version of the AWS plugin (v0.6.0). You may also need to update your AWS CLI v2 version to the latest one as we found some older versions had issues with the AWS SDK for Go.

Can you please give it another try and let us know if using SSO creds with Steampipe works for you?

**Comment by iamjoeker** [comment]
I'm having the same issue with using `aws-vault` which after obtaining credentials from either Keychain or STS sets environment variables and runs a command or spawns a new shell.

My Steampipe aws plugin config is empty which per the docs says it should use the normal sdk resolution order looking first to the environment. aws CLI and other sdk tools consume these creds without issue. Steampipe returns the same as OP. I've not yet figured out how to set the referenced setting for verbose messaging.

**Comment by e-gineer** [comment]
@iamjoeker thanks for trying steampipe and reporting this! I’m not sure why we are not picking up those creds for you, we’ll take a look into this as well.

**Comment by LalitTurbot** [comment]
@iamjoeker  
I have tried to explore a bit around it.

I am the latest version(`0.6.0`) of the Steampipe AWS plugin.
Please use `steampipe plugin update aws` to update to the latest version of the AWS plugin.

Here 
```shell
$ aws-vault add lalit
Enter Access Key Id: ABDCDEFDASDASF
Enter Secret Key: %%%

$ aws-vault exec lalit -- env | grep AWS
AWS_DEFAULT_REGION=us-east-1
AWS_VAULT=vault1
AWS_REGION=us-east-1
AWS_ACCESS_KEY_ID=dummy_access_id
AWS_SECRET_ACCESS_KEY=dummy_secret_key
AWS_SESSION_TOKEN=dummy_session_token
AWS_SECURITY_TOKEN=dummy_session_token
AWS_SESSION_EXPIRATION=2021-03-06T14:58:05Z

$ export AWS_ACCESS_KEY_ID=dummy_access_id
$ export AWS_SECRET_ACCESS_KEY=dummy_secret_key
$ export AWS_SESSION_TOKEN=dummy_session_token

➜  ~ steampipe query                        
Welcome to Steampipe v0.2.3
For more information, type .help

> select name from aws.aws_s3_bucket
+-------------------------------------------------+
| name                                            |
+-------------------------------------------------+
| aws-test-bucket-123                             |
+-------------------------------------------------+

> select * from aws.aws_iam_user
Error: InvalidClientTokenId: The security token included in the request is invalid
	status code: 403, request id: eb263f03-65cc-45d9-9e4d-bd408cd6ed25
```
Seems like it is working for tables which doesn't use AWS IAM service.

https://github.com/99designs/aws-vault/blob/master/USAGE.md#temporary-credentials-limitations-with-sts-iam

Please let us know If we should try a different way to replicate the problem.

**Comment by LalitTurbot** [comment]
@iamjoeker 
can you please share details of  `~/.steampipe/aws.spc`

**Comment by iamjoeker** [comment]
@LalitTurbot Below:

```
connection "aws" {
  plugin    = "aws"
}
```

Although...Good news to report. I updated the aws plugin to 0.8.0 (0.6.0 was the latest as of my original post) and now everything is working as expected. Aws-vault with STS role assumption, session, and just access keys all working so far.

Thanks so much for the troubleshooting assistance!

**Comment by e-gineer** [comment]
Awesome time hear we're running smoothly with SSO and aws-vault in particular. Thanks @iamjoeker for confirming we've got this issue nailed 👍. We appreciate your help with the troubleshooting!

**Comment by cbruno10** [comment]
@jweyrich I've created a new issue for you in https://github.com/turbot/steampipe-plugin-aws/issues/1127, as this is a better place to discuss AWS plugin issues, and I don't necessarily want to re-open this thread as the original issue was different than what you're encountering.
